bson/src/main/org/bson/codecs/UuidCodec.java

```diff
@@ -44,23 +45,35 @@
      * @see org.bson.UuidRepresentation
      */
     public UuidCodec(final UuidRepresentation uuidRepresentation) {
-        this.encoderUuidRepresentation = uuidRepresentation;
-        this.decoderUuidRepresentation = uuidRepresentation;
+        notNull("uuidRepresentation", uuidRepresentation);
+        this.uuidRepresentation = uuidRepresentation;
     }
 
     /**
      * The constructor for UUIDCodec, default is JAVA_LEGACY
      */
     public UuidCodec() {
-        this.encoderUuidRepresentation = UuidRepresentation.JAVA_LEGACY;
-        this.decoderUuidRepresentation = UuidRepresentation.JAVA_LEGACY;
+        this.uuidRepresentation = UuidRepresentation.JAVA_LEGACY;
+    }
+
+    /**
+     * The {@code UuidRepresentation} with which this instance is configured
+     *
+     * @return the uuid representation
+     * @since 3.12
+     */
+    public UuidRepresentation getUuidRepresentation() {
+        return uuidRepresentation;
     }
 
     @Override
     public void encode(final BsonWriter writer, final UUID value, final EncoderContext encoderContext) {
-        byte[] binaryData = UuidHelper.encodeUuidToBinary(value, encoderUuidRepresentation);
+        if (uuidRepresentation == UuidRepresentation.UNSPECIFIED) {
+            throw new BsonSerializationException(""); // TODO: good message
```

Comment:
TODO! Perhaps something like:

> The UUID Representation has not been specified, so the UUID cannot be encoded.

Also should that be a `CodecConfigurationException`?

Reply:
Yes, I think so.  Done, and updated tests to match.